### PR TITLE
feat (explorer): add wrapper designation to order details

### PR DESCRIPTION
**Pull request by kaze-cow**
# Summary

Renders a "Wrappers" row in the verbose order details table for orders whose `appData` references known wrapper contracts. Wrapper-specific UI is code-split and loaded on demand; unknown addresses fall back to fetching the contract `name()` on-chain (supported by all `CowWrapper` contracts).

Supported wrappers for starting off: Euler EVC – Open Position, Close Position, Collateral Swap'

Open Position (https://explorer-dev-git-explorer-wrapper-designation-cowswap-dev.vercel.app/orders/0xea2c93014d0a41820e3b49f7273790041bfd0829e0a26841ab8c10cddfb0d55d9fa3c00a92ec5f96b1ad2527ab41b3932efeda5869eb4e11)

localhost_4200_orders_0xea2c93014d0a41820e3b49f7273790041bfd0829e0a26841ab8c10cddfb0d55d9fa3c00a92ec5f96b1ad2527ab41b3932efeda5869eb4e11

Collateral Swap (https://explorer-dev-git-explorer-wrapper-designation-cowswap-dev.vercel.app/orders/0xef7e3cca05ecddad3515ea524c9be16995fa05824b23e624bd0993fb3ad9842c9fa3c00a92ec5f96b1ad2527ab41b3932efeda5869eb62fe)

localhost_4200_orders_0xef7e3cca05ecddad3515ea524c9be16995fa05824b23e624bd0993fb3ad9842c9fa3c00a92ec5f96b1ad2527ab41b3932efeda5869eb62fe

Close Position (https://explorer-dev-git-explorer-wrapper-designation-cowswap-dev.vercel.app/orders/0x85f93c3fe29fc35e966c36145a997e2a254d7098fe5ebff928e5c09c917fc45dd6cfba01838223dbb47956a2f40d701ce70c382869ef6355)

localhost_4200_orders_0x85f93c3fe29fc35e966c36145a997e2a254d7098fe5ebff928e5c09c917fc45dd6cfba01838223dbb47956a2f40d701ce70c382869ef6355

Unknown wrapper (https://explorer-dev-git-explorer-wrapper-designation-cowswap-dev.vercel.app/orders/0x5594c109c44dd585bc7c494998e1c06dce1b42dfae3dc0fd38094fa4b04130b1e227398f066f11e7fa49893bb97cac6caa7790f969d3acc3)

localhost_4200_orders_0x5594c109c44dd585bc7c494998e1c06dce1b42dfae3dc0fd38094fa4b04130b1e227398f066f11e7fa49893bb97cac6caa7790f969d3acc3

Mobile version:

localhost_4200_orders_0x85f93c3fe29fc35e966c36145a997e2a254d7098fe5ebff928e5c09c917fc45dd6cfba01838223dbb47956a2f40d701ce70c382869ef6355 (1)

# To Test

1. Obtain a Order ID for a trade that uses Euler. This address has a ton of Euler trades:

2. Start the explorer with `pnpm start:explorer`

3. Load the page on the cow explorer. Observe.

- [ ] Looks visually appealing
- [ ] Errors in the display do not bubble up to destroy the whole page

# Background

As wrappers are expected to proliferate in the future, we will want to display useful information on the explorer to aid users (and ourselves as devs) in understanding an order.

Please let me know if the overall strategy for displaying wrapper information seems reasonable. The idea is to make it easy to plug in new wrapper informations when the time comes while causing minimal impact on the rest of the repo

fixes https://linear.app/cowswap/issue/SC-63/add-to-explorer

## Summary by CodeRabbit

* **New Features**
  * Shows wrapper execution details in order info and adds Euler position displays (open/close/collateral swap) in order details.
  * Resolves and displays vault/token metadata and contract-derived values for richer vault views.

* **Documentation**
  * Expanded app conventions guide with UI, linking, caching, and typing standards.

* **Chores**
  * Added an LRU-bounded in-memory cache utility to improve runtime performance.

* **Tests**
  * Expanded LRU utility tests to cover ordering and eviction edge cases.

### CODE REVIEW

**Comment by kaze-cow** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by fairlighteth** [comment]
Found a few regressions / polish items to address before merge:

  1. `explorer:lint` fails on PR-touched files. Main issues are import ordering, missing explicit return type in `_vaultLookup.ts`, and complexity over limit in `ClosePositionComponent` / `CollateralSwapComponent`. Please fix manually.

  2. AGENTS rule violations: `getOrderWrappers.ts` uses `entry.address.toLowerCase()` for registry lookup; please use `getAddressKey(entry.address)` from `@cowprotocol/cow-sdk`. `_vaultLookup.ts` also uses non-null assertions like `chainId!` / `shares!`; those should be removed with proper guards.

  3. Historical wrapper metadata can show the wrong omittable state. Older schemas used `is_omittable`, latest uses `isOmittable`; current code only reads `isOmittable`. Suggested fallback: `entry.isOmittable ?? entry.is_omittable ?? false`.

  4. Wrapper rows are keyed only by `wrapper.address`, so repeated wrapper addresses with different data can collide in React reconciliation. Use a stable key that includes index/data.

  5. The subaccount/owner muted text is very hard to read. It currently uses `Color.explorer_textSecondary` (`#545454`); `Color.explorer_grey` (`#8D8DA9`) would match the muted text used elsewhere better.

  6. “Empty Wrapper” is a misleading fallback label for unknown wrappers. “Empty” sounds like the wrapper does nothing, but it may simply be unrecognized or undecodable by Explorer. Prefer “Unknown wrapper” / “Unrecognized wrapper”, and only use “No wrapper data” when calldata is actually empty.

  7. For unknown wrappers, it would help to add a tooltip explaining why details are missing, e.g. Explorer does not recognize this wrapper address yet, so it cannot decode wrapper-specific details. This may be a custom wrapper or a supported wrapper missing from the registry.

**Review by fairlighteth — COMMENTED** [review]
Flagged some comments. Please review.

File: apps/explorer/src/utils/wrappers/_vaultLookup.ts [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      createPublicClient({
+        chain: VIEM_CHAINS[chainId],
+        transport: http(RPC_URLS[chainId]),
+      }),
+    )
+  }
+  return clientCache.get(chainId)!
+}
+
+const ERC4626_ABI = [
```
Comment (limitofzero) [inline apps/explorer/src/utils/wrappers/_vaultLookup.ts]: nitpick: can you pls move abis to separate const file?

File: apps/explorer/src/utils/wrappers/eulerEvc.tsx [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+}
+
+// ─── Open Position ────────────────────────────────────────────────────────────
+
+const OPEN_POSITION_ABI = parseAbiParameters([
+  '(address owner, address account, uint256 deadline, address collateralVault, address borrowVault, uint256 collateralAmount, uint256 borrowAmount) params',
+  'bytes signature',
+])
+
+export function OpenPositionComponent({ data }: { data: string }): React.ReactElement | null {
```
Comment (limitofzero) [inline apps/explorer/src/utils/wrappers/eulerEvc.tsx]: nitpick: it will be grate to move components per different files (1 component per file)

File: apps/explorer/src/utils/getOrderWrappers.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+export function getOrderWrappers(fullAppData: string | undefined): ResolvedWrapper[] {
+  const appData = decodeFullAppData(fullAppData)
+  if (!appData) return []
+
+  const metadata = (appData as { metadata?: cowAppDataLatestScheme.Metadata }).metadata
+  const entries = metadata?.wrappers ?? []
+
+  return entries.map((entry) => {
+    const registryEntry = WRAPPERS_BY_ADDRESS[entry.address.toLowerCase()]
```
Comment (limitofzero) [inline apps/explorer/src/utils/getOrderWrappers.ts]: use getAddressKey from cow sdk to get address instead of toLowerCase

File: apps/explorer/src/utils/getOrderWrappers.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      return {
+        address: entry.address,
+        data: entry.data,
+        isOmittable: entry.isOmittable ?? false,
+        info: registryEntry.info,
+        loadComponent: registryEntry.loadComponent,
+      }
+    }
+
+    const short = `${entry.address.slice(0, 6)}…${entry.address.slice(-4)}`
```
Comment (limitofzero) [inline apps/explorer/src/utils/getOrderWrappers.ts]: we have an util: shortenAddress

File: apps/explorer/src/utils/wrappers/_vaultLookup.ts [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+  {
+    name: 'debtOf',
+    type: 'function',
+    stateMutability: 'view',
+    inputs: [{ name: 'account', type: 'address' }],
+    outputs: [{ type: 'uint256' }],
+  },
+] as const
+
+export function useConvertToAssets(
```
Comment (limitofzero) [inline apps/explorer/src/utils/wrappers/_vaultLookup.ts]: nitpick: would be grate to move hooks in separate files(1 hook per file).

File: apps/explorer/src/utils/wrappers/_vaultLookup.ts [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+        abi: VAULT_DEBT_ABI,
+        functionName: 'debtOf',
+        args: [account as `0x${string}`],
+      }) as Promise<bigint>,
+  )
+
+  return data
+}
+
+export function useContractName(address: string): string | undefined {
```
Comment (limitofzero) [inline apps/explorer/src/utils/wrappers/_vaultLookup.ts]: nitpick: use AddressKey alias for addresses

File: apps/explorer/src/utils/wrappers/eulerEvc.tsx [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+    <a href={url} target="_blank" rel="noopener noreferrer" title={tokenAddress}>
+      {symbol}
+    </a>
+  )
+}
+
+function OwnerLink({ address }: { address: string }): React.ReactElement {
+  return (
+    <Link to={`/address/${address}`} title={address}>
+      {address.slice(0, 6)}…{address.slice(-4)}
```
Comment (limitofzero) [inline apps/explorer/src/utils/wrappers/eulerEvc.tsx]: nitpick:  shortenAddress

File: apps/explorer/src/utils/wrappers/unknown.tsx [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+
+import { useContractName } from './_vaultLookup'
+
+export function createUnknownComponent(address: string): React.ComponentType<{ data: string }> {
+  function UnknownWrapperComponent(_props: { data: string }): React.ReactElement | null {
```
Comment (limitofzero) [inline apps/explorer/src/utils/wrappers/unknown.tsx]: nitpick: we use direct import ComponentType, ReactElement
for cases like ReactElement | null you can write ReactNode

File: apps/explorer/src/utils/wrappers/_vaultLookup.ts [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+import useSWR from 'swr'
+import { createPublicClient, http } from 'viem'
+
+import { orderBookApi } from '../../sdk/cowSdk'
+import { useNetworkId } from '../../state/network/hooks'
+
+const clientCache = new Map<number, ReturnType<typeof createPublicClient>>()
+
+function getPublicClient(chainId: SupportedChainId) {
+  if (!clientCache.has(chainId)) {
```
Comment (limitofzero) [inline apps/explorer/src/utils/wrappers/_vaultLookup.ts]: nitpick: I think it's better to store this data in jotai/atom

File: apps/explorer/src/utils/wrappers/_vaultLookup.ts [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+    clientCache.set(
+      chainId,
+      createPublicClient({
+        chain: VIEM_CHAINS[chainId],
+        transport: http(RPC_URLS[chainId]),
+      }),
+    )
+  }
+  return clientCache.get(chainId)!
```
Comment (limitofzero) [inline apps/explorer/src/utils/wrappers/_vaultLookup.ts]: we try to avoid ! type assertion

File: apps/explorer/src/utils/wrappers/orderFlowFactory.tsx [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+  'uint256 buyAmount',
+  'bytes32 appData',
+  'uint256 feeAmount',
+  'uint32 validTo',
+  'bool partiallyFillable',
+  'int64 quoteId',
+])
+
+export function OrderFlowFactoryComponent({ data }: { data: string }): React.ReactElement | null {
+  let decoded: ReturnType<typeof decodeAbiParameters<typeof ABI_PARAMS>>
```
Comment (limitofzero) [inline apps/explorer/src/utils/wrappers/orderFlowFactory.tsx]: nitpick: let decoded = decodeAbiParameters(ABI_PARAMS, data as `0x${string}`);
if (!decoded) return null;

File: apps/explorer/src/utils/wrappers/_vaultLookup.ts [hunk 12]
```diff
@@ -0,0 +1,201 @@
...
+    chainId && vaultAddress ? `vault-asset:${chainId}:${vaultAddress}` : null,
+    () =>
+      getPublicClient(chainId!).readContract({
+        address: vaultAddress as `0x${string}`,
+        abi: ERC4626_ABI,
+        functionName: 'asset',
+      }) as Promise<string>,
+  )
+
+  const { data: symbol } = useSWR<string>(
```
Comment (limitofzero) [inline apps/explorer/src/utils/wrappers/_vaultLookup.ts]: it's better to use multicall here

```
const { data: erc20Data } = useSWR(
    assetAddress && chainId ? `erc20-meta:${chainId}:${assetAddress}` : null,
    () =>
      getPublicClient(chainId!).multicall({                                  
        contracts: [                                                       
          { address: assetAddress as `0x${string}`, abi: ERC20_ABI,          
  functionName: 'symbol' },                                                  
          { address: assetAddress as `0x${string}`, abi: ERC20_ABI,
  functionName: 'name' },                                                    
          { address: assetAddress as `0x${string}`, abi: ERC20_ABI,        
  functionName: 'decimals' },                                                
        ],
      }),                                                                    
  )
```

**Review by limitofzero — DISMISSED** [review]

**Comment by limitofzero** [comment]
@coderabbitai review

**Comment by limitofzero** [comment]
There also lint issues that should be addressed

**Comment by anxolin** [comment]
In the description, its nicer for reviewers to provide the links to the explorer link from this PR
https://explorer-dev-git-explorer-wrapper-designation-cowswap-dev.vercel.app/orders/0xea2c93014d0a41820e3b49f7273790041bfd0829e0a26841ab8c10cddfb0d55d9fa3c00a92ec5f96b1ad2527ab41b3932efeda5869eb4e11

Not sure if it's easy, but it could be useful to provide a link to the position in Euler website

**Comment by kaze-cow** [comment]
> Historical wrapper metadata can show the wrong omittable state. Older schemas used is_omittable, latest uses isOmittable; current code only reads isOmittable. Suggested fallback: entry.isOmittable ?? entry.is_omittable ?? false.

The historical wrapper metadata was never, in practice, actually used, and was only specified in error for a very short time when wrappers hadn't yet been established. Is there a way we can just that schema use entirely? (would be a separate PR though)

> “Empty Wrapper” is a misleading fallback label for unknown wrappers. “Empty” sounds like the wrapper does nothing, but it may simply be unrecognized or undecodable by Explorer. Prefer “Unknown wrapper” / “Unrecognized wrapper”, and only use “No wrapper data” when calldata is actually empty.

This is as intended! All wrappers expose a `name()` function so when the explorer doesn't know what the wrapper is by address, it can determine by querying on-chain so that there is something to display, and in this case, the wrapper is literally called "Empty Wrapper" and wouldn't normally be what is displayed for other wrappers. Sorry for the confusion.

Rest of the changes from that post addressed in b9c2b7303 @fairlighteth . Still working on other comments. Thanks for review!

File: apps/explorer/src/utils/getOrderWrappers.ts (hunk 3 above)
Comment (kaze-cow) [reply apps/explorer/src/utils/getOrderWrappers.ts]: b9c2b7303

File: apps/explorer/src/utils/wrappers/_vaultLookup.ts (hunk 1 above)
Comment (kaze-cow) [reply apps/explorer/src/utils/wrappers/_vaultLookup.ts]: part of 5ae040056

File: apps/explorer/src/utils/wrappers/eulerEvc.tsx (hunk 2 above)
Comment (kaze-cow) [reply apps/explorer/src/utils/wrappers/eulerEvc.tsx]: part of 5d6a34880

File: apps/explorer/src/utils/getOrderWrappers.ts (hunk 3 above)
Comment (kaze-cow) [reply apps/explorer/src/utils/getOrderWrappers.ts]: part of https://github.com/cowprotocol/cowswap/pull/7404/commits/b9c2b730351e6ca4fa25bdc36043b11112e880ac

File: apps/explorer/src/utils/wrappers/_vaultLookup.ts (hunk 5 above)
Comment (kaze-cow) [reply apps/explorer/src/utils/wrappers/_vaultLookup.ts]: primary fix of [5ae0400](https://github.com/cowprotocol/cowswap/pull/7404/commits/5ae040056cf9e453205af35b404ef90e8534ac1e)

File: apps/explorer/src/utils/wrappers/unknown.tsx (hunk 8 above)
Comment (kaze-cow) [reply apps/explorer/src/utils/wrappers/unknown.tsx]: this file was completely removed.

**Comment by kaze-cow** [comment]
> Not sure if it's easy, but it could be useful to provide a link to the position in Euler website

Once the position is closed/recycled, its no longer possible to view it from anywhere. and they don't really do the whole thing with explorer outside of https://www.explorer.euler.finance/ , which seems more vault data focused. I figure the order ID is easily extracted, though.

Will update the other links.

File: apps/explorer/src/utils/wrappers/_vaultLookup.ts (hunk 10 above)
Comment (kaze-cow) [reply apps/explorer/src/utils/wrappers/_vaultLookup.ts]: fixed as part of [b9c2b73](https://github.com/cowprotocol/cowswap/pull/7404/commits/b9c2b730351e6ca4fa25bdc36043b11112e880ac)

File: apps/explorer/src/utils/wrappers/unknown.tsx (hunk 8 above)
Comment (kaze-cow) [reply apps/explorer/src/utils/wrappers/unknown.tsx]: found other instances of the issue and resolved/replaced them.

Comment (kaze-cow) [reply apps/explorer/src/utils/wrappers/unknown.tsx]: fixed in 43e4b0780

File: apps/explorer/src/utils/wrappers/_vaultLookup.ts (hunk 12 above)
Comment (kaze-cow) [reply apps/explorer/src/utils/wrappers/_vaultLookup.ts]: fixed in 43e4b0780

File: apps/explorer/src/utils/getOrderWrappers.ts (hunk 4 above)
Comment (kaze-cow) [reply apps/explorer/src/utils/getOrderWrappers.ts]: fixed 43e4b0780

File: apps/explorer/src/utils/wrappers/_vaultLookup.ts (hunk 6 above)
Comment (kaze-cow) [reply apps/explorer/src/utils/wrappers/_vaultLookup.ts]: fixed 43e4b0780

File: apps/explorer/src/utils/wrappers/eulerEvc.tsx (hunk 7 above)
Comment (kaze-cow) [reply apps/explorer/src/utils/wrappers/eulerEvc.tsx]: fixed 43e4b0780

**Comment by elena-zh** [comment]
Hey @kaze-cow , should the PR still be in draft or it is 'ready to review'?

**Comment by kaze-cow** [comment]
> Hey @kaze-cow , should the PR still be in draft or it is 'ready to review'?

Almost there! I am just wrapping up a couple final comments, fixing the CI, and then it will be ready.

**Review by elena-zh — COMMENTED** [review]
Overall. changes LGTM

I've updated order links in the PR description: changed the URL to the pr preview instead of the localhost. 

Some nitpicks; 
1. Links styles for addresses and tokens should be similar everywhere in the app 

Image

2. It would be nice to wrap sentences by words on a smaller screen resolutions: 

Image

File: apps/explorer/src/utils/wrappers/_vaultLookup.ts (hunk 9 above)
Comment (kaze-cow) [reply apps/explorer/src/utils/wrappers/_vaultLookup.ts]: fixed in 62109a83e and aff025cd0

File: apps/explorer/src/utils/wrappers/orderFlowFactory.tsx (hunk 11 above)
Comment (kaze-cow) [reply apps/explorer/src/utils/wrappers/orderFlowFactory.tsx]: this file was removed (should not have been included in the first place; this particular wrapper contract is saved for later)

**Comment by kaze-cow** [comment]
by the way in order to satisfy the suggested requirement of , I ended up including `jotai-family` because the `jotai/utils` [implementation of `atomFamily` is apparently deprecated](https://jotai.org/docs/utilities/family).

However, this seems to have introduced a lot of unexpected lockfile changes, so much so that it appears to be affecting the swap website build.

If anyone from the team can help and rebuild the lockfile properly, that would be much appreciated :pray: 

Additionally, `Vercel - storybook` job appeasr to [already be broken in develop](https://github.com/cowprotocol/cowswap/commit/1f13c9760300af7d06653d052116ec72c4bb530e), so I am not going t worry about it.

**Review by elena-zh — CHANGES_REQUESTED** [review]
Hey @kaze-cow , I think it is up to @fairlighteth  to decide, however, I'd reduce tokens size z bit to look like this: 
image


Also, this [issue ](https://github.com/cowprotocol/cowswap/pull/7404#pullrequestreview-4235855280) (case 2) has not been addressed 
image

**Comment by kaze-cow** [comment]
> Also, this https://github.com/cowprotocol/cowswap/pull/7404#pullrequestreview-4235855280(case 2) has not been addressed

@elena-zh try once more, looks like I forgot to push that change :/

**Review by elena-zh — APPROVED** [review]
LGTM!
As for the design, will leave it up to @fairlighteth to decide :)

File: pnpm-lock.yaml
Comment (Danziger) [inline pnpm-lock.yaml:1]: Regarding the changes here, I don't think this is an issue. We have a PR ready to pin the versions, so this will soon be more stable: https://github.com/cowprotocol/cowswap/pull/7450

File: apps/explorer/src/components/orders/OrderWrapperDetails/index.tsx
Comment (Danziger) [inline apps/explorer/src/components/orders/OrderWrapperDetails/index.tsx:1]: OrderWrapperDetails.tsx might be a better name.

See https://www.notion.so/cownation/CoW-Swap-React-Dapp-Architecture-1858da5f04ca806b9796cfd63c982275.

File: apps/explorer/src/components/orders/OrderWrapperDetails/item/OrderWrapperDetailsItem.styled.tsx
Comment (Danziger) [inline apps/explorer/src/components/orders/OrderWrapperDetails/item/OrderWrapperDetailsItem.styled.tsx:1]: OrderWrapperDetails.styled.tsx

File: apps/explorer/src/utils/wrappers/eulerEvc/closePosition.tsx [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+        </TokenBlock>
+      </TradeCard>
+      <SubInfo>
+        Subaccount #{sub} · Owner <OwnerLink address={params.owner} />
+      </SubInfo>
+    </div>
+  )
+}
+
+export function ClosePositionComponent({ data }: { data: string }): ReactElement | null {
```
Comment (Danziger) [inline apps/explorer/src/utils/wrappers/eulerEvc/closePosition.tsx]: Ideally, create 2 different files and export 1 component from each of them, with a matching name. Also, avoid the "Component" suffix. I suggest:

- ClosePositionComponent => ClosePosition in ClosePosition.tsx
- ClosePositionView in ClosePositionView.tsx

File: apps/explorer/src/utils/wrappers/eulerEvc/positions/EulerPosition.styles.tsx
Comment (Danziger) [inline apps/explorer/src/utils/wrappers/eulerEvc/positions/EulerPosition.styles.tsx:1]: This one should probably be broken down into individual files.

File: apps/explorer/src/utils/wrappers/eulerEvc/openPosition.tsx [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+
+import { VaultAsset, useConvertToAssets, useVaultAsset } from '../../../hooks/euler'
+import { useNetworkId } from '../../../state/network/hooks'
+
+const OPEN_POSITION_ABI = parseAbiParameters([
+  '(address owner, address account, uint256 deadline, address collateralVault, address borrowVault, uint256 collateralAmount, uint256 borrowAmount) params',
+  'bytes signature',
+])
+
+type OpenPositionParams = {
```
Comment (Danziger) [inline apps/explorer/src/utils/wrappers/eulerEvc/openPosition.tsx]: Prefer `interface`.

**Comment by Danziger** [comment]
@kaze-cow @elena-zh I've added a bunch of comments but I'll take care of them in ~1h and make sure the PR is merged today.

**Comment by Danziger** [comment]
> by the way in order to satisfy the suggested requirement of , I ended up including `jotai-family` because the `jotai/utils` [implementation of `atomFamily` is apparently deprecated](https://jotai.org/docs/utilities/family).
> 
> However, this seems to have introduced a lot of unexpected lockfile changes, so much so that it appears to be affecting the swap website build.
> 
> If anyone from the team can help and rebuild the lockfile properly, that would be much appreciated 🙏
> 
> Additionally, `Vercel - storybook` job appeasr to [already be broken in develop](https://github.com/cowprotocol/cowswap/commit/1f13c9760300af7d06653d052116ec72c4bb530e), so I am not going t worry about it.

@kaze-cow I removed `jotai-family` because this cache is not reactive (`getPublicClient` only needs a per-chain singleton memoization, and nothing subscribes to updates). Caching at the module level with `Map<SupportedChainId, PublicClient>` is simpler, but I've introduced `FiniteMap` which is just a wrapper around `Map` with a max number of entries and a LRU eviction strategy just in case. Otherwise this might eventually cause memory issues (if we ever have many chains and forget about this cache).

In any case, `jotai-family` will be introduced in another PR I'm working on, but it has some downsides as requiring us to manage clean up of old atoms manually with a really limitted API.

File: apps/explorer/src/utils/wrappers/eulerEvc/positions/collateral/EulerCollateralSwap.pure.tsx [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+
+export interface EulerCollateralSwapParams {
+  owner: string
+  account: string
+  deadline: bigint
+  fromVault: string
+  toVault: string
+  fromAmount: bigint
+  disableSourceCollateral: boolean
+}
```
[the remark this answers is not in the document]
Comment (Danziger) [reply apps/explorer/src/utils/wrappers/eulerEvc/positions/collateral/EulerCollateralSwap.pure.tsx:32]: @limitofzero I'm not sure if this is right. Are we supposed to use `AddressKey` everywhere and call `getAddressKey()` for each address? The name and TSDoc there might be a bit missleading, so if an address is supposed to be just that, a plain address, and not used as a key, does this guideline still apply?

**Review by Danziger — APPROVED** [review]

**Review by fairlighteth — APPROVED** [review]
